src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1903,4 +1908,64 @@ private Owner createOwnerFromDTO(OwnerDTO ownerDTO) {
         sink.emitOwnerCreated(owner);
         return owner;
     }
+
+    // PUT /owners/{owner_key}/consumers/environments
+
+    @Override
+    @Transactional
+    public void setConsumersToEnvironments(@Verify(Owner.class) String ownerKey,
+        SetConsumerEnvironmentsDTO request) {
+
+        List<String> consumerUuids = request.getConsumerUuids();
+        if (consumerUuids == null || consumerUuids.isEmpty()) {
+            throw new BadRequestException(i18n.tr("Consumer UUIDs is null or empty"));
```

Comment:
Any user-facing messages should be less terse than our usual messaging we leave for ourselves. I would expect this to be something closer to "No consumer UUIDs provided".

Reply:
Updated